draft-packages/filter-drawer/KaizenDraft/FilterDrawer/FilterDrawer.tsx

```diff
@@ -0,0 +1,98 @@
+import { MenuContent } from "@kaizen/draft-menu"
+import { StatelessMenu } from "@kaizen/draft-menu/KaizenDraft/Menu/StatelessMenu"
+import classnames from "classnames"
+import * as React from "react"
+const filterIcon = require("@kaizen/component-library/icons/filter.icon.svg")
+  .default
+const styles = require("./styles.module.scss")
+import { FilterDrawerButton } from "./FilterDrawerButton"
+
+export interface FilterDrawerProps {
+  /**
+   * The text that goes inside the filter button
+   */
+  labelText: string
+
+  /**
+   * Metadata shows relevant info about the content being filtered
+   */
+  metadata?: React.ReactNode
+
+  /**
+   * The content that appears in the filter dropdown.
+   */
+  children: React.ReactElement
+  reversed?: boolean
+
+  /**
+   * The filter drawer does not hold internall state, so you
+   * must pass in the state of whether the dropdown is open
+   * or not
+   */
+  isDropdownVisible: boolean
+
+  /**
+   * You must provide a function that toggles the isDropdownVisible state
+   */
+  toggleDropdown: () => void
+
+  /**
+   * You must provide a function that sets the isDropdownVisible state to `false`
+   */
+  hideDropdown: () => void
+
+  /**
+   * If at least one filter is enabled, then a filter badge will show.
+   * The filter button badge shows the number of filter fields with applied filters.
+   */
+  numberOfFiltersEnabled?: number
+}
+
+const dropdownId = "filter-drawer-dropdown"
```

Comment:
Consider when there are multiple dropdowns on the same page - this would result in multiple identical ids. This will potentially mess with the ariaControls in this component as it won't know which component the id is in reference to. 

I would suggest looking into something like [cuid](https://github.com/ericelliott/cuid) to generate unique ids for this component.

Reply:
That's a good idea.

Reply:
Good call - this was a lazy oversight on my behalf. I decided to go with having the consumer provide the id. I think there is some benefit in human readable ids, or at least letting the consumer decide.